idrstream/CP_idr.py

```diff
@@ -266,9 +270,13 @@ def compile_batch_CP_features(self, output_path: pathlib.Path):
             "Metadata_Well_Number",
         ]
 
-        # remove unnecessary metadata columns
+        # Identify which unneeded columns are present in cp_output
+        columns_to_drop = [col for col in unneeded_columns if col in cp_output.columns]
+
+        # Remove identified unnecessary columns
```

Comment:
I really like this change! I didn't realize how hardcoded I made it before haha. I will likely keep this in mind for the future.